common/notify_stream/src/lib.rs

```diff
@@ -83,6 +84,22 @@ pub enum Error {
     MaxFilesWatch,
 }
 
+#[derive(Debug)]
+pub struct ContextError {
+    error: Error,
+    #[allow(dead_code)]
+    context: Backtrace,
+}
+
+impl ContextError {
+    pub fn new(error: Error) -> Self {
+        Self {
+            error,
+            context: Backtrace::capture(),
+        }
+    }
+}
```

Comment:
Had to create custom error type and constructor to pass the backtrace information along.